src/coreclr/vm/jitinterface.cpp

```diff
@@ -10614,6 +10614,39 @@ void* CEEJitInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,         /* IN  */
         }
 #endif
 
+        // Array element accessors are more perf sensitive than other managed helpers and indirection
+        // costs introduced by PreStub could be noticeable (7% to 30% depending on platform).
+        // Other helpers are either more complex, less common, or have their trivial case inlined by the JIT.
+        //
+        // So if these helpers already reached Tier1 somewhere, we can safely skip the PreStub indirection.
+        if (dynamicFtnNum == DYNAMIC_CORINFO_HELP_ARRADDR_ST || dynamicFtnNum == DYNAMIC_CORINFO_HELP_LDELEMA_REF)
+        {
+            Precode* pPrecode = Precode::GetPrecodeFromEntryPoint((PCODE)hlpDynamicFuncTable[dynamicFtnNum].pfnHelper);
+            _ASSERTE(pPrecode->GetType() == PRECODE_FIXUP);
+
+            MethodDesc* helperMD = pPrecode->GetMethodDesc();
+            _ASSERT(helperMD != nullptr);
+
+            NativeCodeVersion::OptimizationTier tier;
+
+            {
+                CodeVersionManager::LockHolder codeVersioningLockHolder;
```

Comment:
We should have a static array that caches the Tier1 pointers of the JIT helpers so that we do not have to take this lock all the time.